cpp/src/ffi_go/ir/LogTypes.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef FFI_GO_IR_LOG_TYPES_HPP
+#define FFI_GO_IR_LOG_TYPES_HPP
+
+#include <string>
+#include <vector>
+
+#include <clp/components/core/src/Defs.h>
+#include <clp/components/core/src/ffi/encoding_methods.hpp>
+
+#include <ffi_go/LogTypes.hpp>
+
+namespace ffi_go::ir {
+
+template <typename>
+[[maybe_unused]] constexpr bool cAlwaysFalse{false};
+
+template <typename encoded_var_t>
+struct LogMessage {
+    auto reserve(size_t cap) -> void { m_logtype.reserve(cap); }
```

Comment:
Shall we be more explicit that it reserves the storage for `m_logtype`?
```suggestion
    auto reserve_logtype(size_t cap) -> void { m_logtype.reserve(cap); }
```

Reply:
This is done to mirror std:string